src/ws/transport.rs

```diff
@@ -458,15 +463,27 @@ async fn per_connection_task(
                     })
                     .now_or_never();
 
-                if let Some(Ok(_)) = result {
-                    pending_requests.push(request_id);
-                } else {
-                    // TODO: why are we terminating all pending requests if only one fails?!.
-                    log::error!(
-                        "Send request={:?} failed; terminating all pending_requests",
-                        request_id
-                    );
-                    return pending_requests;
+                match result {
+                    // Request was succesfully transmitted to the frontend.
+                    Some(Ok(_)) => pending_requests.push(request_id),
+                    // The channel is down or full.
+                    Some(Err(e)) => {
+                        log::error!(
+                            "send request={:?} to frontend failed because of {:?}, terminating the connection",
+                            request_id,
+                            e,
+                        );
+                        return pending_requests;
+                    }
+                    // The future wasn't ready.
+                    // TODO(niklasad1): verify if this is possible to happen "in practice".
```

Comment:
I'm unsure if this is the right action if the `future` wasn't ready then we terminate the connection currently.

It might be possible that the receiving side ("frontend") is busy doing other things so it might be better just to ignore the request then notify the client and continue process the connection further.